handbook/engineering/security/index.md

```diff
@@ -4,50 +4,39 @@ We think that security is an enabler for the business. Sourcegraph is committed
 
 ## Goals
 
-### Long term
+Our current work is documented in our [tracking issue](https://github.com/sourcegraph/sourcegraph/issues?q=is%3Aissue+label%3Atracking+label%3Ateam%2Fsecurity+is%3Aopen).
 
-Every organization is confident they can trust Sourcegraph cloud to securely store and manage their source code.
+### Visibility into Sourcegraph Cloud's attack surface
 
-### Medium term
+**Problem:** The Cloud team is working toward [private code on Sourcegraph Cloud](../cloud/index.md#private-code-on-sourcegraph-cloud). Before we can have [confidence in our security model](#confidence-in-our-security-model), we first need to have visibility into current risks, issues, and threats, so we can proactively plan what work needs to be done (i.e., we don't want to be relying on or waiting for security researchers or customer to report vulnerabilities to us).
 
-The goals below lie on the path to our long-term goals, and represent the next 3-6 months of work.  These goals will be delivered serially, but may be worked on in parallel.
+**Milestones:**
 
-**We host our private [infrastructure repository](http://github.com/sourcegraph/infrastructure) on Sourcegraph cloud**
+1. All Docker images are continuously scanned for known security vulnerabilities and the security team is alerted any time one is found.
+1. All compute nodes are continuously scanned for known security vulnerabilities and the security team is alerted any time one is found.
+   - GCP doesn't handle this?
+1. All high risk (I mean this informally, is there a better term of art to use here?) vulnerabilities found in scans are resolved.
+1. Implement centralized security logging.
+   - What is this? Is this just taking all our existing logs and shipping them to one place? Are there new kinds of logs we will want to add to our application? If so, what/where? We are already using LogDNA, is that a solution to this?
```

Comment:
We should not be using LogDNA — that was an experiment that @keegancsmith started. I think we should keep our logs, which may contain sensitive data, inside our core infrastructure (i.e GCP), so we should use https://console.cloud.google.com/logs/viewer?project=sourcegraph-dev

Reply:
We wanted store security logs (to start) in Hosted ElasticSearch to get started. The other side of the same discussion is storing (also Elastic backed) in our own infrastructure, but creating a new GCP project separate from production and dev.

Reply:
I think we need to align on one centralized logging solution for engineering at large (including security use-cases), so this needs to be discussed. Will there be an RFC?

Reply:
yeah please link me to this rfc. As a sidenote we have had pretty bad experiences with elastic search for logging in the past. Currently (and for quite a while) I imagine the amount of log data we generate can just be searched with grep and we can avoid solutions that require maintaining an index.

Reply:
Can I ask what the issues were that were experienced in the past? Was it speed, of search, managing Elastic itself, of something else entirely?

Reply:
All 3. It was slow, it was tricky to update configuration for it to parse our structured logs and it would constantly die. All in all I don't understand using it for logs at our scale (or even much larger scales). If you are generating a few GB of logs a day, that is totally reasonable to just grep. We also won't be using extra features kibana provides like alerting, we should be sticking to prometheus for that.

Maybe someone with more experience it it could make it work well for us, but to be honest this seems like a problem that shouldn't require someone with specific experience :P

Reply:
So we used it to build a large data pipeline for security logs, as did the folks over at [siemmonster](https://siemonster.com/community-edition/).  We had it up to about 100GB of logs a day across, with closing indices, and structuring. But, we were using logstash to prep the data in the first place.

I fully agree - Elastic experience makes a difference in making it behave rationally.  However, the security side does care about the alerting. Our goal/use case is to both have the centralized logs, and work down the path of correlation.

Reply:
Implementation details should be discussed in a relevant RFC when we get to that milestone. For now, it is enough to say that we want to have a place where we can search over all of our logs.